GitCommands/PathUtil.cs

```diff
@@ -124,14 +124,42 @@ public static string NormalizePath([NotNull] this string path)
 
             try
             {
-                return Path.GetFullPath(new Uri(path).LocalPath);
+                return Path.GetFullPath(Resolve(path));
             }
             catch (UriFormatException)
             {
                 return string.Empty;
             }
         }
 
+        [NotNull]
+        public static string Resolve([NotNull] string path, string relativePath = "")
+        {
+            if (string.IsNullOrWhiteSpace(path))
+            {
+                throw new ArgumentException(nameof(path));
+            }
+
+            // Fix for on purpose invalid machine name of WSL in Windows 10
+            bool replaceWslPath = path.StartsWith(@"\\wsl$\");
```

Comment:
Is `wsl$` always in the lower case?
Can it be `WSL$` or `Wsl$`?

Reply:
It's not case sensitive, I will fix it.

Reply:
Use [this overload](https://docs.microsoft.com/dotnet/api/system.string.startswith?view=netframework-4.6.1#System_String_StartsWith_System_String_System_StringComparison_) then

Reply:
Since it is case insensitive, it makes the handling slightly different.
Instead of regex, use `string.Substring` with the known length.